### ISSUE TITLE
getData() is returning data.frame of 0 observations

### ISSUE THREAD

**Issue by davidbmitchell** [beanumber/openWAR#70]
After running the following code, the function returned a data.frame that was 8.5mb but had 0 observations.
```r
MLBAM2015 <- getData(start="2015-04-05", end="2015-06-08")
```
It printed out the gameIds like it normally does, and you couldn't tell something was wrong until you try to run the ```makeWAR``` function.
```r
ds <- makeWAR(MLBAM2015)

 Error in `$<-.data.frame`(`*tmp*`, "idx", value = c(1L, 0L)) : 
  replacement has 2 rows, data has 0 
```
Could this be caused by the refactoring of dplyr's NSE?

**Comment by gjm112** [comment]
I'm going to try to reproduce this.

On Tue, Jun 9, 2015 at 11:47 AM, David B. Mitchell <notifications@github.com
> wrote:

> After running the following code, the function returned a data.frame that
> was 8.5mb but had 0 observations.
>
> MLBAM2015 <- getData(start="2015-04-05", end="2015-06-08")
>
> It printed out the gameIds like it normally does, and you couldn't tell
> something was wrong until you try to run the makeWAR function.
>
> ds <- makeWAR(MLBAM2015)
>
>  Error in `$<-.data.frame`(`*tmp*`, "idx", value = c(1L, 0L)) :
>   replacement has 2 rows, data has 0
>
> Could this be caused by the refactoring of dplyr's NSE?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/beanumber/openWAR/issues/70>.
>



-- 
Gregory J. Matthews, Ph.D.
Assistant Professor
Department of Mathematics and Statistics
Loyola University Chicago
E-mail: gjm112 -at- gmail.com
Blog: StatsInTheWild.com
Art: etsy.me/1JAsYz9
Twitter: @StatsInTheWild <http://www.twitter.com/StatsInTheWild>
Twitter: @StatsClass <http://www.twitter.com/statsclass>

**Comment by gjm112** [comment]
I get the same thing.  This is a big problem.  I will try to fix this now.

**Comment by gjm112** [comment]
Line 47 is where is breaks.

**Comment by davidbmitchell** [comment]
I see where that could break it. How do we fix it. Can we just change it to: 
```r
out = dplyr::filter_(out, ~game_type == "R")
```
dplyr's NSE and SE confuses me big time.

**Comment by davidbmitchell** [comment]
After making the above change, I do believe that fixes it. I just tried and it works again. I ran:
```r
MLBAM2015 <- getData(start = "2015-04-05", end = "2015-04-07")
```
and it returned a ```data.frame``` of 1581 obs.

**Comment by gjm112** [comment]
I just pushed a fix.  Can you try pulling and see if it works.  

Why do we need the ~ in front of game_type?

**Comment by gjm112** [comment]
And what is the difference between filter and filter_???

**Comment by gjm112** [comment]
Looks like Ben made this change to get the package to pass R CMD check.

Will my fix not work?  Is yours better?

On Tue, Jun 9, 2015 at 12:10 PM, David B. Mitchell <notifications@github.com
> wrote:

> After making the above change, I do believe that fixes it. I just tried
> and it works again. I ran:
>
> MLBAM2015 <- getData(start = "2015-04-05", end = "2015-04-07")
>
> and it returned a data.frame of 1581 obs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/beanumber/openWAR/issues/70#issuecomment-110436531>.
>



-- 
Gregory J. Matthews, Ph.D.
Assistant Professor
Department of Mathematics and Statistics
Loyola University Chicago
E-mail: gjm112 -at- gmail.com
Blog: StatsInTheWild.com
Art: etsy.me/1JAsYz9
Twitter: @StatsInTheWild <http://www.twitter.com/StatsInTheWild>
Twitter: @StatsClass <http://www.twitter.com/statsclass>

**Comment by davidbmitchell** [comment]
It will, I made mine before I was aware that you were working on one as well. The only difference with mine is when running ```R CMD CHECK``` it won't fuss about global variable bindings. The difference between ```filter``` and ```filter_``` is that ```filter``` uses dplyr's non standard evaluation syntax and causes ```R CMD CHECK``` to throw up a note. Using ```filter_``` is the standard evaluation syntax and ```R CMD CHECK``` does not fuss. Its really convoluted and makes little sense me.

**Comment by davidbmitchell** [comment]
See issue #52 where I add some links that provide more explanation regarding the differences.

**Comment by gjm112** [comment]
So you are saying that your changes are better than mine?  I will accept
your changes then because we want to pass R CMD check.

On Tue, Jun 9, 2015 at 12:24 PM, David B. Mitchell <notifications@github.com
> wrote:

> See issue #52 <https://github.com/beanumber/openWAR/issues/52> where I
> add some links that provide more explanation regarding the differences.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/beanumber/openWAR/issues/70#issuecomment-110439369>.
>



-- 
Gregory J. Matthews, Ph.D.
Assistant Professor
Department of Mathematics and Statistics
Loyola University Chicago
E-mail: gjm112 -at- gmail.com
Blog: StatsInTheWild.com
Art: etsy.me/1JAsYz9
Twitter: @StatsInTheWild <http://www.twitter.com/StatsInTheWild>
Twitter: @StatsClass <http://www.twitter.com/statsclass>

**Comment by gjm112** [comment]
How do I accept your changes?

On Tue, Jun 9, 2015 at 12:31 PM, Gregory Matthews <gjm112@gmail.com> wrote:

> So you are saying that your changes are better than mine?  I will accept
> your changes then because we want to pass R CMD check.
>
> On Tue, Jun 9, 2015 at 12:24 PM, David B. Mitchell <
> notifications@github.com> wrote:
>
>> See issue #52 <https://github.com/beanumber/openWAR/issues/52> where I
>> add some links that provide more explanation regarding the differences.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/beanumber/openWAR/issues/70#issuecomment-110439369>.
>>
>
>
>
> --
> Gregory J. Matthews, Ph.D.
> Assistant Professor
> Department of Mathematics and Statistics
> Loyola University Chicago
> E-mail: gjm112 -at- gmail.com
> Blog: StatsInTheWild.com
> Art: etsy.me/1JAsYz9
> Twitter: @StatsInTheWild <http://www.twitter.com/StatsInTheWild>
> Twitter: @StatsClass <http://www.twitter.com/statsclass>
>
>


-- 
Gregory J. Matthews, Ph.D.
Assistant Professor
Department of Mathematics and Statistics
Loyola University Chicago
E-mail: gjm112 -at- gmail.com
Blog: StatsInTheWild.com
Art: etsy.me/1JAsYz9
Twitter: @StatsInTheWild <http://www.twitter.com/StatsInTheWild>
Twitter: @StatsClass <http://www.twitter.com/statsclass>

**Comment by davidbmitchell** [comment]
That's what I'm not sure about because there is a merge conflict.

**Comment by gjm112** [comment]
I made your change in my code and then pushed it to github.  (I gave you credit in the commit note.)

**Comment by gjm112** [comment]
Thanks for alerting us to this by the way.  We keep breaking this thing.

**Comment by davidbmitchell** [comment]
Its all good, I am happy to help. I absolutely love this project. We keep on breaking it because it's not mature yet, but it is getting there. We're learning more and more, so I think its beneficial to the project in the long run.

**Comment by gjm112** [comment]
We appreciate all the help we can get.

On Tue, Jun 9, 2015 at 12:40 PM, David B. Mitchell <notifications@github.com
> wrote:

> Its all good, I am happy to help. I absolutely love this project. We keep
> on breaking it because it's not mature yet, but it is getting there. We're
> learning more and more, so I think its beneficial to the project in the
> long run.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/beanumber/openWAR/issues/70#issuecomment-110443044>.
>



-- 
Gregory J. Matthews, Ph.D.
Assistant Professor
Department of Mathematics and Statistics
Loyola University Chicago
E-mail: gjm112 -at- gmail.com
Blog: StatsInTheWild.com
Art: etsy.me/1JAsYz9
Twitter: @StatsInTheWild <http://www.twitter.com/StatsInTheWild>
Twitter: @StatsClass <http://www.twitter.com/statsclass>

**Comment by beanumber** [comment]
This is fixed now, right? 

Here is my best understanding of what happened. It all has to do with non-standard evaluation and @hadley's use of it in `dplyr`. 

What we used to have was:

```
out = dplyr::filter(out, game_type == "R")
```
This works, but makes use of [non-standard evaluation](http://cran.r-project.org/web/packages/dplyr/vignettes/nse.html). The question is how does `filter()` know that `game_type` is a column in the `out` data frame, and not an object in the workspace? Generally, if you write something without quotes, then R looks for it in the workspace. But that is not what R is doing here, because Hadley wrote `filter()` to use non-standard evaluation. While this will work just fine, `R CMD check` doesn't like non-standard evaluation, and so it issues a NOTE, which we don't want. 

Of course, @hadley thought of this, and so he wrote `filter_()`, which works exactly the same way as `filter()`, except that it doesn't do non-standard evaluation. So now, what type of object is the second argument to `filter()`? 

I changed the code to:
```
out = dplyr::filter_(out, "game_type" == "R")
```
This fixed the NOTE, since now it's clear that `game_type` and `R` are both literals, but this was dumb because it doesn't work at all. The literal `game_type` is **never** equal to the literal `R`, so that's why the data frame returned 0 rows. I think it still takes up memory because it's a `tbl` and not just a `data.frame`.

The correct fix is @davidbmitchell 's:
```
out = dplyr::filter_(out, ~game_type == "R")
```
The presence of the `~` forces R to recognize the second argument as a `formula`, which is what we want. 

Incidentally, we now have to go through all of the instances of the `dplyr` verbs and add an `_` and a `~`.
